### PR TITLE
Creates course through api call

### PR DESCRIPTION
**Pull request by techievivek**
API endpoint to create course.

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -429,4 +431,63 @@ def get(self, request, answerpaper_id, format=None):
...
+                        return Response(unit_serializer.errors)
+                module_instance.learning_unit.set(units)
+            else:
+                return Response(module_serializer.errors)
+        if serializer.is_valid():
+            new_course = serializer.save()
+            new_course.learning_module.set(modules)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        else:
+            return Response(serializer.errors)
```
Comment (adityacp) [inline api/views.py:493]: Send a 400 status code if there errors in serializers. 400 is for bad requests.

File: online_test/settings.py [hunk 2]
```diff
@@ -215,19 +215,19 @@
...
+#     'DEFAULT_AUTHENTICATION_CLASSES': [
+#         'rest_framework.authentication.TokenAuthentication',
+#         'rest_framework.authentication.BasicAuthentication',
+#         'rest_framework.authentication.SessionAuthentication',
+#     ],
+#     'DEFAULT_PERMISSION_CLASSES': [
+#         'rest_framework.permissions.IsAuthenticated'
+#     ],
+#     'TEST_REQUEST_DEFAULT_FORMAT': 'json'
+# }
```
Comment (adityacp) [inline online_test/settings.py:230]: Don't comment this.

File: yaksh/migrations/0018_comment_forumbase_post.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Generated by Django 3.0.3 on 2020-05-02 08:35
```
Comment (adityacp) [inline yaksh/migrations/0018_comment_forumbase_post.py:1]: Remove all the migration files. They are not needed.

File: api/views.py [hunk 4]
```diff
@@ -429,4 +431,63 @@ def get(self, request, answerpaper_id, format=None):
...
+        for learning_module in request.data['learning_module']:
+            learning_module['creator'] = user_id
+            for learning_unit in learning_module["learning_unit"]:
+                if learning_unit['type'] == 'quiz':
+                    learning_unit['quiz']['creator'] = user.id
+                if learning_unit['type'] == 'lesson':
+                    learning_unit['lesson']['creator'] = user.id
+        serializer = CourseSerializer(data=request.data)
+        modules = []
+        for learning_module in request.data['learning_module']:
```
Comment (adityacp) [inline api/views.py:452]: You can do the serializer part in the previous for loop only where you are setting creator, no need of adding another for loop.

File: .env [hunk 5]
```diff
@@ -1,11 +1,11 @@
 # Django settings
 SECRET_KEY=dUmMy_s3cR3t_k3y
-#DB_ENGINE=mysql
-#DB_NAME=yaksh
-#DB_USER=root
-#DB_PASSWORD=root
+DB_ENGINE=mysql
```
Comment (adityacp) [inline .env:3]: Don't rename .samplenv to .env this is not required it will be done in the production.

File: api/tests.py [hunk 6]
```diff
@@ -904,3 +904,38 @@ def add(a,b):
...
+            'name': 'Python Test Api Course',
+            'creator': 'demo',
+            'enrollment': 'Enroll Request',
+            'learning_module': leaningModule
+        }
+        response = self.client.post(reverse('api:course_create'), data)
+        # Then
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertTrue(
+            Course.objects.filter(name='Python Test Api Course').exists())
```
Comment (adityacp) [inline api/tests.py:937]: Check if the learning module and quiz also exists or not.

File: api/tests.py [hunk 7]
```diff
@@ -904,3 +904,38 @@ def add(a,b):
...
+        LearningUnit = {'quiz': quiz, 'type': 'quiz', 'order': 1}
+        leaningModule = {
+            'name': 'LM1',
+            'description': 'module one',
+            'creator': 'demo',
+            'learning_unit': LearningUnit
+        }
+        data = {
+            'name': 'Python Test Api Course',
+            'creator': 'demo',
```
Comment (adityacp) [inline api/tests.py:929]: why are you sending a `creator` in the post data for the quiz, learning module?
The creator is required only at the course level.

**Comment by adityacp** [comment]
Your test cases are failing in the Travis ci. Fix the test

File: api/tests.py (hunk 7 above)
Comment (techievivek) [reply api/tests.py:929]: LearningModule too have a creator field.

Comment (adityacp) [reply api/tests.py:929]: I know but anyway creator will be added in the views for the learning module and quiz, so you don't need it in the post data.

**Pull request closed without merging**